src/scheduler/MpiWorld.cpp

```diff
@@ -448,6 +477,9 @@ void MpiWorld::recv(int sendRank,
     std::shared_ptr<faabric::MPIMessage> m =
       getLocalQueue(sendRank, recvRank)->dequeue();
 
+    if (!m) {
+        throw std::runtime_error("Recevied a malformed message");
+    }
```

Comment:
Does this ever happen? The message in this case isn't malformed it's null. I would just lose this if it never happens, or add a more detailed error log line if it does (e.g. Recv message 1 -> 2 is null)